File: akka-javasdk/src/test/scala/akka/javasdk/impl/agent/MyModelProvider.scala
/*
 * Copyright (C) 2021-2025 Lightbend Inc. <https://www.lightbend.com>
 */

package akka.javasdk.impl.agent

import akka.javasdk.agent.ModelProvider
import com.typesafe.config.Config

class MyModelProvider(config: Config) extends ModelProvider.Custom {

  override def modelName(): String = {
    config.getString("model-name")
  }

  override def createChatModel(): AnyRef = ???

  override def createStreamingChatModel(): AnyRef = ???
}

File: samples/doc-snippets/src/main/java/com/example/application/MyModelProvider.java
package com.example.application;

import akka.javasdk.agent.ModelProvider;
import com.typesafe.config.Config;

// tag::custom[]
public class MyModelProvider implements ModelProvider.Custom {

  private final String modelName;

  public MyModelProvider(Config config) { // <1>
    modelName = config.getString("model-name");
    //TODO model provider initialization based on injected config
  }

  @Override
  public String modelName() {
    return modelName;
  }

  @Override
  public Object createChatModel() { // <2>
    //TODO return ChatModel implementation
    return null;
  }

  @Override
  public Object createStreamingChatModel() { //<3>
    //TODO return StreamingChatModel implementation
    return null;
  }
}
// end::custom[]
